mas-foundation/src/main/java/com/ca/mas/core/policy/PolicyManager.java

```diff
@@ -101,9 +140,15 @@ private void processResponse(RequestInfo request, MASResponse response, List<Mss
     }
 
     public MASResponse execute(RequestInfo requestInfo, Route<MASResponse> function) throws MAGException, MAGServerException, IOException {
-        processRequest(requestInfo, defaultPolicy);
+
+        boolean whichPolicy = true;
+        if (requestInfo.getExtra() != null && Boolean.parseBoolean((String) requestInfo.getExtra().get(LOGOUT_EXTRA))) {
```

Comment:
Make it generic, use the request url to determine which policy to run, we don't need LOGOUT_EXTRA in this case.